dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeValidityAndRenewTest.java

```diff
@@ -0,0 +1,355 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.acme.fat;
+
+import static junit.framework.Assert.assertNotNull;
+import static org.junit.Assert.assertNotSame;
+
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.security.acme.docker.CAContainer;
+import com.ibm.ws.security.acme.internal.util.AcmeConstants;
+import com.ibm.ws.security.acme.utils.AcmeFatUtils;
+
+import componenttest.annotation.CheckForLeakedPasswords;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Verify that we log the correct warnings for different
+ * validity/renewBeforeExpiration settings
+ */
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class AcmeValidityAndRenewTest {
+
+	@Server("com.ibm.ws.security.acme.fat.simple")
+	public static LibertyServer server;
+
+	private static ServerConfiguration ORIGINAL_CONFIG;
+	private static final String[] DOMAINS1 = { "domain1.com" };
+
+	public static CAContainer caContainer;
+	
+	public static final long timeBufferToExpire = 30000L; // milliseconds
+
+	@BeforeClass
+	public static void beforeClass() throws Exception {
+		ORIGINAL_CONFIG = server.getServerConfiguration();
+		try {
+			caContainer = new com.ibm.ws.security.acme.docker.pebble.PebbleContainer();
+		} catch (Exception e) {
+			caContainer = new com.ibm.ws.security.acme.docker.pebble.PebbleContainer();
```

Comment:
Why is this needed?

Reply:
Oops, it's not -- I was having problems on my box with the Pebble container not starting. Will remove.